libs/template/templates/dbt-sql/template/{{.project_name}}/resources/{{.project_name}}_job.yml.tmpl

```diff
@@ -26,8 +22,13 @@ resources:
             profiles_directory: dbt_profiles/
             commands:
             - 'dbt deps --target=${bundle.target}'
-            - 'dbt seed --target=${bundle.target} --vars "{ dev_schema: {{$dev_schema}} }"'
-            - 'dbt run --target=${bundle.target} --vars "{ dev_schema: {{$dev_schema}} }"'
+{{- if (regexp "^yes").MatchString .personal_schemas}}
+            - 'dbt seed --target=${bundle.target} --vars "{ dev_schema: ${workspace.current_user.short_name} }"'
+            - 'dbt run --target=${bundle.target} --vars "{ dev_schema: ${workspace.current_user.short_name} }"'
```

Comment:
This will also apply to the prod target, which contradicts the explanation in the template:
> yes, use a schema based on the current user name during development

You could use target overrides to only apply this for the dev target.

Reply:
This should actually work as intended since the "dev_schema" var is only used in the dev profile: https://github.com/databricks/cli/pull/1463/files#diff-0ca9390ffa1ff34b28194f7947d477dd79b02f9fe6aa14a578483469913e4d9aR21.

This is the place where our dev/prod targets and the dbt profiles come together. Which can look a bit weird. To make the logic of the generated yml seem bit simpler, I'm no longer passing "dev_schema" unless `yes, use a schema based on the current user name during development` is actually enabled. Once that is enabled, customers will have to stare at the slightly more complicated ymkl.

Reply:
It is possible to do this exclusively for the dev target using overrides. Then it is more clear the dev schema only pertains to the dev target. It's pretty confusing as is, IMO.

```yaml
targets:
  dev:
    resources:
      jobs:
        dbt_project_job:
          tasks:
            - task_key: dbt
              dbt_task:
                commands:
                  - 'dbt deps --target=${bundle.target}'
                  - 'dbt seed --target=${bundle.target} --vars "{ dev_schema: ${workspace.current_user.short_name} }"'
                  - 'dbt run --target=${bundle.target} --vars "{ dev_schema: ${workspace.current_user.short_name} }"'

  prod:
    resources:
      jobs:
        dbt_project_job:
          tasks:
            - task_key: dbt
              dbt_task:
                commands:
                  - 'dbt deps --target=${bundle.target}'
                  - 'dbt seed --target=${bundle.target}'
                  - 'dbt run --target=${bundle.target}'

```

The top level resource then shouldn't specify commands, as lists are appended in overrides.

Reply:
Or define another variable that interpolates the inner string for `--vars`.

Reply:
The challenge with overrides is that it duplicates things and breaks visual editing. And in this case, we're just passing the value of a parameter, not changing behavior. So I'd rather just keep a single version of this job.

Reply:
How about a comment then? The generated configuration should pre-empt questions people have. Maybe "the dev schema is not used in the prod target" or something?

Reply:
Yes that sounds good, added!